src/includes/class-wordlift-countries.php

```diff
@@ -702,17 +321,119 @@ class Wordlift_Countries {
 		'zw' => 'Zimbabwe',
 	);
 
+	/**
+	 * Parse_country_code_json_file_to_array.
+	 *
+	 * @param  string $file_name The json file name where the supported country
+	 * and languages are present.
+	 *
+	 * @return array An Array having two maps, country_code_language_map and country_code_name_map.
+	 */
+	public static function parse_country_code_json_file_to_array( $file_name ) {
+		$json_file_contents = file_get_contents( $file_name );
+		$decoded_array      = json_decode( $json_file_contents, true );
+		// decoded array would be null if the json_decode parses
+		// invalid content.
+		if ( null === $decoded_array ) {
+			return array(
+				'country_code_name_map'     => array(),
+				'country_code_language_map' => array(),
+			);
+		} else {
+			$result = array();
+			// country_code => country_language map.
+			$country_code_language_map = array();
+			// country_code => country_name map.
+			$country_code_name_map = array();
+			foreach ( $decoded_array as $key => $value ) {
+				$country_code_language_map[ $key ] = $value['supportedLang'];
+				$country_code_name_map [ $key ]    = $value['defaultLoc']['loc_name'];
+			}
+			$result['country_code_language_map'] = $country_code_language_map;
+			$result['country_code_name_map']     = $country_code_name_map;
+			return $result;
+		}
+	}
+	/**
+	 * Get the list of WordLift's supported country codes from json file mapping country_code => languages.
+	 *
+	 * @since 3.22.5.1
+	 *
+	 * @param string $file_name The json file where the supported country codes and language_codes are stored.
+	 *
+	 * @return array An Array having two maps, country_code_language_map and country_code_name_map.
+	 */
+	public static function get_codes_from_json_file( $file_name ) {
+		if ( file_exists( $file_name ) ) {
+			return self::parse_country_code_json_file_to_array( $file_name );
+		} else {
+			return array(
+				'country_code_name_map'     => array(),
+				'country_code_language_map' => array(),
+			);
+		}
+	}
+
+	/**
+	 * Returns the country language pairs.
+	 *
+	 * @since 3.18.0
+	 *
+	 * @return array The country language pairs.
+	 */
+	public static function get_codes() {
+		return self::$codes;
+	}
+	/**
+	 * Populate self::codes and self::country_codes if not done before.
+	 *
+	 * @since 3.22.5.1
+	 *
+	 * @param string $file_name The json file where the supported country codes and language_codes are stored.
+	 *
+	 * @return void
+	 */
+	private static function lazy_populate_codes_and_country_codes_array( $file_name ) {
+		if ( null === $file_name ) {
+			$file_name = __DIR__ . '/supported-countries.json';
```

Comment:
@naveen17797 should we use `plugin_dir_path( dirname( __FILE__ ) )` instead of `__DIR__`?

Reply:
__DIR__ and dirname( __FILE__ ) produces exactly same output, but __DIR__ seems to be slightly faster than dirname( __FILE__ ) 
Benchmark Results: https://beeznest.com/blog/2015/02/17/benchmark-dirname-vs-dir-php/
since PHP 5.3 is the minimum supported version we could use this magic constant __DIR__ (if it is confusing, and plugin_dir_path() explicitly specifies the file is in the plugin directory we could use that , i am not sure about what to use,  it depends upon your decision ).